src/components/Configure/content/fields/FieldMappings/RequiredFieldMappings.tsx

```diff
@@ -31,10 +34,23 @@ export function RequiredFieldMappings() {
   };
 
   const integrationFieldMappings = useMemo(
-    () => configureState?.read?.requiredMapFields?.filter(
-      isIntegrationFieldMapping,
-    ) || [],
-    [configureState],
+    () => {
+      if (!selectedObjectName || !fieldMapping) return [];
```

Comment:
Can you add some comments here on this transformation to make this easier to maintain?